launch/launch/launcher.py

```diff
@@ -39,6 +40,11 @@ def __init__(self, name_prefix='', sigint_timeout=10):
         self.sigint_timeout = sigint_timeout
         self.task_descriptors = []
         self.print_mutex = threading.Lock()
+        self.loop = None
+        self.loop_lock = threading.Lock()
+        self.interrupt_future = asyncio.Future()
+        self.launch_complete = threading.Event()
+        self.processes_launched = threading.Event()
```

Comment:
By similar logic, you could argue the launch function should be spawn, but I think this makes sense to change: 32a840c